src/ami-set.js

```diff
@@ -0,0 +1,73 @@
+let base = require('taskcluster-base');
+let debug = require('debug')('aws-provisioner:AmiSet');
+
+const KEY_CONST = 'ami-set';
+
+ /**
+ * An AMI Set is a collection of AMIs with a single name (its AMI Set ID).
+ * Each AMI in the set is keyed by its virtualization type (PV or HVM) and
+ * by its AWS region.
+ */
+
+let AmiSet = base.Entity.configure({
+  version: 1,
+
+  partitionKey: base.Entity.keys.ConstantKey(KEY_CONST),
+  rowKey: base.Entity.keys.StringKey('id'),
+
+  properties: {
+
+    id: base.Entity.types.String,
+    /* This is a JSON object which contains the AMIs of an AMI set keyed by
+     * their virtualization type and region. It is in the shape:
+     * [
+     *   {
+     *     region: us-west-1,
+     *     hvm: ami-2222
+     *     pv: ami-3333
+     *   }
+     * ]
+     */
+    amis: base.Entity.types.JSON,
+    // Store the date of last modification for this entity
+    lastModified: base.Entity.types.Date,
+
+  },
+});
+
+/**
+ * Load all the knwon Amis
+ */
+AmiSet.listAmiSets = async function () {
+
+  let amiSetList = [];
+
+  try {
+    await base.Entity.scan.call(this, {}, {
+      handler: function(item) {
+        amiSetList.push(item.json());
```

Comment:
I believe we already had some discussion of this in the PR?  Many of the list methods in TC services return the entire object, e.g., [auth.listClients](https://docs.taskcluster.net/reference/platform/auth/api-docs#listClients).  The reason *not* to do so is if it requires additional queries or a lot more bandwidth.  Neither is the case here.  We have the additional information about each AMI set returned from Azure, so why not provide it to the user?